DS13/code/modules/turbolift/turbolift.dm

```diff
@@ -0,0 +1,234 @@
+
+/area/turbolift/secondary
+	name = "Secondary turbolift"
+
+/area/turbolift/tertiary
+	name = "Tertiary turbolift"
+
+/area/turbolift/quanery
+	name = "Quanery turbolift"
+
+/area/turbolift/quintessential
+	name = "Quintessential turbolift"
+
+//Structures and logic//
+
+/obj/structure/turbolift
+	name = "Turbolift control console"
+	icon = 'DS13/icons/obj/decor/wall_decor.dmi'
+	icon_state = "lift-off"
+	density = FALSE
+	anchored = TRUE
+	can_be_unanchored = FALSE
+	mouse_over_pointer = MOUSE_HAND_POINTER
+	desc = "Starfleet's decision to replace the iconic turboladder was not met with unanimous praise, experts citing increased obesity figures from crewmen no longer needing to climb vertically through several miles of deck to reach their target. However this is undoubtedly much faster."
+	var/floor_directory = "<font color=blue>Deck 1: Engineering <br>\
+		Deck 2: Promenade<br>\
+		Deck 3: Bridge<br></font>" //Change this if you intend to make a new map. Helps players know where they're going.
+	var/list/turbolift_turfs = list()
+	var/floor = 0 //This gets assigned on init(). Allows us to calculate where the lift needs to go next.
+	var/list/destinations = list() //Any elevator that's on our path.
+	pixel_y = 32 //This just makes it easier for locate...
+	var/in_use = FALSE //Is someone using a lift? If they are, then don't let anyone in the lift.
+	var/max_floor = 0 //Highest floor we can go to?
+	var/bolted = FALSE //Is this door bolted or unbolted? do we need to change that if a person walks into our lift?
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+
+/obj/machinery/door/airlock/trek/goon/turbolift
+	name = "Turbolift airlock"
+	icon = 'DS13/goonstation/glass_airlock.dmi'
+	desc = "A sleek airlock for walking through. This one looks extremely strong."
+	icon_state = "closed"
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF | FREEZE_PROOF
+
+/turf/closed/wall/trek_smooth/indestructible
+	name = "Duotanium alloy hull"
+	icon = 'DS13/icons/turf/smooth_trek_walls.dmi'
+	desc = "A large chunk of metal used to seperate rooms, decks and even build structures in space."
+	icon_state = "wall"
+	smooth = TRUE
+	legacy_smooth = TRUE
+	canSmoothWith = list(/turf/closed/wall/trek,/turf/closed/wall/trek/darksteel,/turf/closed/wall/trek/room,/obj/structure/window/reinforced/fulltile/trek,/obj/structure/window/reinforced/fulltile/trek/viewport,/obj/structure/window/reinforced/fulltile/trek/corridor,/obj/machinery/door/airlock/trek,/obj/machinery/door/airlock/trek/tng)
+	explosion_block = 50
+
+/turf/closed/wall/trek_smooth/indestructible/TerraformTurf(path, defer_change = FALSE, ignore_air = FALSE)
+	return
+
+/turf/closed/wall/trek_smooth/indestructible/acid_act(acidpwr, acid_volume, acid_id)
+	return 0
+
+/turf/closed/wall/trek_smooth/indestructible/attackby()
+	return FALSE //No deconning them
+
+/obj/structure/turbolift/process()
```

Comment:
DENIED.

Use a system like the teleporter hub where it all connects up and one of them handles all this shit. Along with using a timer to know when they all unbolt.